File: symfony/src/Entity/StudentAnswer.php
<?php

namespace App\Entity;

use App\Repository\StudentAnswerRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: StudentAnswerRepository::class)]
#[ORM\Table(name: 'student_answers')]
class StudentAnswer
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(name: 'specific_score_id')]
    private ?int $specificScoreId = null;

    #[ORM\Column(name: 'user_id')]
    private ?int $userId = null;

    #[ORM\Column(name: 'specific_test_id')]
    private ?int $specificTestId = null;

    #[ORM\Column(name: 'question_id')]
    private ?int $questionId = null;

    #[ORM\Column(name: 'question_text', type: 'text')]
    private ?string $questionText = null;

    #[ORM\Column(name: 'selected_answer_text', type: 'text')]
    private ?string $selectedAnswerText = null;

    #[ORM\Column(name: 'answer_score')]
    private ?int $answerScore = 0;

    #[ORM\Column(name: 'passed_at', type: 'datetime')]
    private ?\DateTimeInterface $passedAt = null;

    public function __construct()
    {
        $this->passedAt = new \DateTime();
    }

    public function getId(): ?int { return $this->id; }
    public function getSpecificScoreId(): ?int { return $this->specificScoreId; }
    public function setSpecificScoreId(int $specificScoreId): static { $this->specificScoreId = $specificScoreId; return $this; }
    public function getUserId(): ?int { return $this->userId; }
    public function setUserId(int $userId): static { $this->userId = $userId; return $this; }
    public function getSpecificTestId(): ?int { return $this->specificTestId; }
    public function setSpecificTestId(int $specificTestId): static { $this->specificTestId = $specificTestId; return $this; }
    public function getQuestionId(): ?int { return $this->questionId; }
    public function setQuestionId(int $questionId): static { $this->questionId = $questionId; return $this; }
    public function getQuestionText(): ?string { return $this->questionText; }
    public function setQuestionText(string $questionText): static { $this->questionText = $questionText; return $this; }
    public function getSelectedAnswerText(): ?string { return $this->selectedAnswerText; }
    public function setSelectedAnswerText(string $selectedAnswerText): static { $this->selectedAnswerText = $selectedAnswerText; return $this; }
    public function getAnswerScore(): ?int { return $this->answerScore; }
    public function setAnswerScore(int $answerScore): static { $this->answerScore = $answerScore; return $this; }
    public function getPassedAt(): ?\DateTimeInterface { return $this->passedAt; }
    public function setPassedAt(\DateTimeInterface $passedAt): static { $this->passedAt = $passedAt; return $this; }
}

File: integrated-project/mindboost-javafx/src/main/java/com/mindboost/app/model/StudentAnswer.java
package com.mindboost.app.model;

import jakarta.persistence.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "student_answers")
public class StudentAnswer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "specific_score_id", nullable = false)
    private Integer specificScoreId;

    @Column(name = "user_id", nullable = false)
    private Integer userId;

    @Column(name = "specific_test_id", nullable = false)
    private Integer specificTestId;

    @Column(name = "question_id", nullable = false)
    private Integer questionId;

    @Column(name = "question_text", columnDefinition = "LONGTEXT", nullable = false)
    private String questionText;

    @Column(name = "selected_answer_text", columnDefinition = "LONGTEXT", nullable = false)
    private String selectedAnswerText;

    @Column(name = "answer_score", nullable = false)
    private Integer answerScore = 0;

    @Column(name = "passed_at", nullable = false)
    private LocalDateTime passedAt = LocalDateTime.now();

    public Integer getId() {
        return id;
    }

    public Integer getSpecificScoreId() {
        return specificScoreId;
    }

    public void setSpecificScoreId(Integer specificScoreId) {
        this.specificScoreId = specificScoreId;
    }

    public Integer getUserId() {
        return userId;
    }

    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    public Integer getSpecificTestId() {
        return specificTestId;
    }

    public void setSpecificTestId(Integer specificTestId) {
        this.specificTestId = specificTestId;
    }

    public Integer getQuestionId() {
        return questionId;
    }

    public void setQuestionId(Integer questionId) {
        this.questionId = questionId;
    }

    public String getQuestionText() {
        return questionText;
    }

    public void setQuestionText(String questionText) {
        this.questionText = questionText;
    }

    public String getSelectedAnswerText() {
        return selectedAnswerText;
    }

    public void setSelectedAnswerText(String selectedAnswerText) {
        this.selectedAnswerText = selectedAnswerText;
    }

    public Integer getAnswerScore() {
        return answerScore;
    }

    public void setAnswerScore(Integer answerScore) {
        this.answerScore = answerScore;
    }

    public LocalDateTime getPassedAt() {
        return passedAt;
    }

    public void setPassedAt(LocalDateTime passedAt) {
        this.passedAt = passedAt;
    }
}
